ros/processors/report_processor.py

```diff
@@ -0,0 +1,110 @@
+            finally:
+                consumer.commit()
+
+        consumer.close()
+
+    def handle_message(self):
+        self.report_url = self.msg.get('url', None)
+        metadata = self.msg.get('metadata', None)
+        if not metadata or not metadata['insights_id']:
+            return None
+        insights_id = metadata['insights_id']
+        rh_identity = self.msg.get('b64_identity', None)
+        host = fetch_host_from_inventory(insights_id, rh_identity)
+        if not host['results']:
+            print("No record found. Make sure system is registered in insights")
+            return None
+        if not self.report_url:
+            print("kafka message missing report url")
+        host_id = host['results'][0]['id']
+        report_tar = self._download_report()
+        performance_record = self._extract_performance_record(report_tar)
+
+        with app.app_context():
+            if performance_record:
+                performance_score = self._calculate_performance_score(performance_record)
+                profile = PerformanceProfile.query.filter_by(inventory_id=host_id).first()
+                if profile:
+                    profile.performance_record = performance_record
+                    profile.performance_score = performance_score
+                else:
+                    record = PerformanceProfile(inventory_id=host_id,
+                                                performance_record=performance_record,
+                                                performance_score=performance_score)
+                    db.session.add(record)
+
+                db.session.commit()
+
+    def _calculate_performance_score(self, performance_record):
+        memory_score = (float(performance_record['avg_memory_used']) / float(performance_record['avg_memory'])) * 100
+        performance_score = {'memory_score': int(memory_score)}
+        return performance_score
+
+    def _download_report(self):
+        download_response = requests.get(self.report_url)
+        if download_response.status_code != HTTPStatus.OK:
+            print("Unable to download the report")
+        return download_response.content
+
+    def _extract_performance_record(self, report_tar):
+        tar = tarfile.open(fileobj=BytesIO(report_tar), mode='r:*')
+        files = tar.getmembers()
+        metrics_file = None
+        for file in files:
+
+            if '/metrics.json' in file.name or file.name == 'metrics.json':
+                metrics_file = file
+
+        if metrics_file:
+            extracted_metrics_file = tar.extractfile(metrics_file)
+            record_string = extracted_metrics_file.read().decode('utf-8')
+            performance_record = json.loads(record_string)
+            return performance_record
+
+
+def check_kafka_connection():
+    topics = consumer.list_topics().topics
+    if "platform.upload.resource-optimization" in topics:
+        return True
+    else:
+        return False
```

Comment:
```suggestion
    return ("platform.upload.resource-optimization" in topics)
```
Using `in`, it returns True if element exists in list and False if the element does not exists in list.